iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/ShuntCompensatorNonLinearModelImpl.java

```diff
@@ -49,19 +49,30 @@ public double getG() {
 
     @Override
     public double getMaximumB() {
-        return sections.values().stream()
-                .map(SectionImpl::getB)
-                .max(Double::compare)
-                .orElseThrow(() -> new PowsyblException("a shunt compensator must have at least one section"));
+        double sumB = sections.values().stream().mapToDouble(SectionImpl::getB).sum();
+        return Math.max(0, sumB);
     }
 
     @Override
     public double getMaximumG() {
+        double sumG = sections.values().stream().mapToDouble(SectionImpl::getG).filter(g -> !Double.isNaN(g)).sum();
+        return Math.max(0, sumG);
+    }
+
+    @Override
+    public double getMinimumB() {
         return sections.values().stream()
-                .map(SectionImpl::getG)
-                .filter(g -> !Double.isNaN(g))
-                .max(Double::compare)
-                .orElse(Double.NaN);
+                .mapToDouble(SectionImpl::getB)
+                .filter(b -> b <= 0)
+                .sum();
+    }
+
+    @Override
+    public double getMinimumG() {
+        return sections.values().stream()
+                .mapToDouble(SectionImpl::getG)
+                .filter(g -> !Double.isNaN(g) && g <= 0)
+                .sum();
     }
```

Comment:
I think it's not correct: the code should be the same than getMaximumB / getMaximumG except this is a min?